spark/dl/src/main/scala/com/intel/analytics/bigdl/utils/DistriParameterSynchronizer.scala

```diff
@@ -39,111 +41,83 @@ trait DistriParameterSynchronizer[T] {
    * @param globalSize total size of parameter
    * @param priority priority for this parameter
    */
-  def init(name: String, globalSize: Int, priority: Int = 1): Unit
+  def init(name: String, globalSize: Int, priority: Int = 1, weights: Tensor[T]
+           , grads: Tensor[T]): Unit
 
   /**
    * put parameter to global
    * @param name identifier for parameter
-   * @param parameter  parameter to put
    */
-  def put(name: String, parameter: Tensor[T]): Unit
+  def put(name: String): Unit
 
   /**
    * get parameter for specific identifier
    * @param name  identifier for parameter
    * @return  parameter
    */
-  def get(name: String): Tensor[T]
+  def get(name: String): (Tensor[T], Tensor[T])
 
   /**
    * clear the synchronizer
    */
   def clear(): Unit
 }
 
-class BlockManagerParameterSynchronizer[T: ClassTag](partitionID: Int,
-                                                     totalPartition: Int)
-                                                    (implicit ev: TensorNumeric[T])
-  extends DistriParameterSynchronizer[T] {
+class BlockManagerParameterSynchronizer[T: ClassTag](val partitionID: Int,
+                                       val totalPartition: Int)
+                                      (implicit ev: TensorNumeric[T])
+  extends DistriParameterSynchronizer[T]{
 
   import com.intel.analytics.bigdl.utils.BlockManagerParameterSynchronizer.logger
 
   @volatile private var shutdown = false
 
+  private val communicationStartCore = CpuInfo.getPhysicalProcessorCount - 4
+
   private val syncResults: mutable.HashMap[String, FutureTask[Tensor[T]]]
-    = new mutable.HashMap[String, FutureTask[Tensor[T]]]()
+  = new mutable.HashMap[String, FutureTask[Tensor[T]]]()
 
   private val taskSize: Int = System.getProperty("bigdl.ParameterSynchronier." +
-    "asyncTaskSize", "100").toInt
+    "asyncTaskSize", "500").toInt
 
   private val clearPoolSize: Int = System.getProperty("bigdl.ParameterSynchronier." +
-    "clearPoolSize", "1").toInt
+    "clearPoolSize", "2").toInt
 
   private val workerPoolSize: Int = System.getProperty("bigdl.ParameterSynchronier" +
-    ".syncPoolSize", "4").toInt
+    ".workerPoolSize", "4").toInt
 
   private val syncPoolSize: Int = Math.max(System.getProperty("bigdl.ParameterSynchronier" +
-    ".computePoolSize",
+    ".syncPoolSize",
     (Runtime.getRuntime().availableProcessors() / 2).toString).toInt, 2)
 
```

Comment:
Need we check the invalid of all these java options ?

Reply:
it should be fine with default values